### PR TITLE
CB-10960 ignore false error in case of freeipa user_mod operation

### PR DESCRIPTION
**Pull request by horadla23**
See detailed description in the commit message.

### CODE REVIEW

**Review by lacikaaa — APPROVED** [review]

**Review by handavid — COMMENTED** [review]
The other operations follow the same pattern as setting the workload credentials. shouldn't the error handling remain in the WorkloadCredentialsService? I think it would be better to do add an `invokeOperation(UserModOperation)` method to the WorkloadCredentialsService. Does that not work with the batching code?

File: freeipa-client/src/main/java/com/sequenceiq/freeipa/client/operation/UserModOperation.java [hunk 1]
```diff
@@ -51,10 +52,19 @@ public String getOperationName() {
...
-        LOGGER.debug("modified user {}", user);
-        return Optional.of(modified);
+        try {
+            LOGGER.debug("modifying user {}", user);
+            User modified = invoke(freeIpaClient, User.class);
+            LOGGER.debug("modified user {}", user);
+            return Optional.of(modified);
+        } catch (FreeIpaClientException e) {
+            if (FreeIpaClientExceptionUtil.isEmptyModlistException(e)) {
+                LOGGER.debug("Workload credentials for user '{}' already set.", getUser());
```
[the remark this answers is not in the document]
Comment (horadla23) [reply freeipa-client/src/main/java/com/sequenceiq/freeipa/client/operation/UserModOperation.java:62]: yes, we use this in every case of user_mod calls, so everywhere we will use this, FreeipaClient also uses this to invoke the operation

**Comment by horadla23** [comment]
@handavid 

> The other operations follow the same pattern as setting the workload credentials. shouldn't the error handling remain in the WorkloadCredentialsService? 

Sometimes operations contain error handling, for example GroupAdd, GroupRemove, GroupAddMember, GroupRemoveMember. When error handling is the same everytime, then the operation class should contain it. If there are different error handling, for example in case of UserAdd (kerberos user creation, ldap user creation, usersync user creation), the error handling should be in the service classes.

> I think it would be better to do add an invokeOperation(UserModOperation) method to the WorkloadCredentialsService. 

I am not sure what would be the benefit. Also user_mod operation is not always related to logics of WorkloadCredentialService, sometimes we are just changing password, or just updating password expiration (SetPasswordHandler, KerberosConfigService, LdapConfigService).

> Does that not work with the batching code?

UserMod also works with the batching code. but we are using it only in case of usersync (and of course if entitlement is on), if there is password change, or password expiration change (those can happen independently from the usersync, as mentioned above) we are using single calls.

**Comment by handavid** [comment]
@horadla23 Thanks for the explanation. I approve of your changes

**Pull request merged**